openfortivpn-webview: init at 1.2.0

lriesebos opened the pull request:
## Description of changes

first contribution: added openfortivpn-webview package.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

aos commented on pkgs/by-name/op/openfortivpn-webview/package.nix:
```diff
@@ -0,0 +1,34 @@
+  inherit pname version;
+
+  src = fetchurl {
+    url = "https://github.com/gm-vm/${pname}/releases/download/v${version}-electron/${pname}-${version}.tar.xz";
```
Since this is a release on Github, you should use the `fetchFromGithub` fetcher instead: https://nix.dev/tutorials/packaging-existing-software.html#fetching-source-from-github

aos reviewed (commented):
1. Use the `fetchFromGithub` fetcher
2. Make sure you follow commit conventions for contributing to nixpkgs: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions, specifically: rename your PR to follow the guideline above, split out the change to add yourself as the maintainer.

lriesebos committed:
maintainers: add lriesebos

Signed-off-by: Leon Riesebos <28567817+lriesebos@users.noreply.github.com>

lriesebos committed:
openfortivpn-webview: init at 1.2.0

Release version v1.2.0-electron

Added the new `--trusted-cert` option to bypass certificate errors.

Signed-off-by: Leon Riesebos <28567817+lriesebos@users.noreply.github.com>

lriesebos commented:
@aos appreciate the feedback for a first-time contributor. sorry for the delay too, just life happened. hope this PR is in a better state now!

aos reviewed (commented):
Making some good progress

lriesebos commented:
@aos , thanks again for the feedback, highly appreciated. let me know if there is anything else!

aos reviewed (commented):
### Edit: Below is incorrect. See: https://github.com/NixOS/rfcs/pull/140, but will keep below for posterity.
<hr />
1. Rename from `package.nix` to `default.nix`
2. You should add the package to `pkgs/top-level/all-packages.nix`, alphabetically.

After these changes, you should be able to build it from the root of the repo with:
```
nix-build -A openfortivpn-webview
```
Once you do - test it out and make sure it works as you expect (perhaps including any version information).

I think that should be everything remaining from my point of view.

lriesebos commented:
@aos , I am a bit confused. the [quick start to adding a package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package) specifically asks for `package.nix`. also, there is no mention of `all-packages.nix` for adding a new package, and the new package builds fine as is. see the output below. recent PR's also seem to point in the same direction (e.g. https://github.com/NixOS/nixpkgs/pull/319908). are you sure your latest change requests are really supposed to be there?

```
leon@tokachi:~/code/nixpkgs$ nix-build -A openfortivpn-webview
/nix/store/9d0ssvxi5ins6as1c1x5r81pkcgp8b5l-openfortivpn-webview-1.2.0
leon@tokachi:~/code/nixpkgs$ nix build .#openfortivpn-webview
leon@tokachi:~/code/nixpkgs$ ls -l result
lrwxrwxrwx. 1 leon leon 70 Jun 15 23:08 result -> /nix/store/9d0ssvxi5ins6as1c1x5r81pkcgp8b5l-openfortivpn-webview-1.2.0
leon@tokachi:~/code/nixpkgs$ ./result/bin/openfortivpn-webview --help
openfortivpn-webview [command]

Commands:
  openfortivpn-webview [host:port]  Connect to the given host

Options:
  --version         Show version number                                [boolean]
  --realm           The authentication realm.                           [string]
  --url             The already built SAML URL. This takes precedence over
                    [host:port].                                        [string]
  --url-regex       A regex to detect the URL that needs to be visited before
                    printing SVPNCOOKIE.
                                      [string] [default: "/sslvpn/portal\.html"]
  --keep-open       Do not close the browser automatically.
  --proxy-server    HTTP Proxy in the format hostname:port.             [string]
  --extra-ca-certs  Path to a file with extra certificates. The file should
                    consist of one or more trusted certificates in PEM format.
                                                                        [string]
  --trusted-cert    The fingerprint of a certificate to always trust, even if
                    invalid. The details of invalid certificates, fingerprint
                    included, will be dumped in the console.            [string]
  --help            Show help                                          [boolean]
```

aos reviewed (approved):
@lriesebos Ah thanks for the correction. Looks like [RFC 140](https://github.com/NixOS/rfcs/pull/140) was merged somewhat recently and the guidelines have also been updated. In that case - disregard my last comment. This is good to go from my point of view :-)

lriesebos commented:
if there is anything else that needs to be done to unblock this, let me know

aos commented:
@lriesebos I would advertise this PR here: https://discourse.nixos.org/t/prs-already-reviewed/2617 or https://discourse.nixos.org/t/prs-ready-for-review-december/1711

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1865

SuperSandro2000 commented on pkgs/by-name/op/openfortivpn-webview/package.nix:
```diff
@@ -0,0 +1,41 @@
+
+let
+  version = "1.2.0";
+in
+
+buildNpmPackage {
+  inherit version;
+
+  pname = "openfortivpn-webview";
```
```suggestion
in
buildNpmPackage {
  pname = "openfortivpn-webview";
  inherit version;
```

lriesebos replied:
thanks for the suggestion, updated that accordingly! also noticed that RFC-166 formatting came through in the meanwhile, so got that reformatted too (after two retries).

lriesebos commented:
@SuperSandro2000 I think I addressed your last concerns. let me know if there is anything else I can do!

SuperSandro2000 commented:
@ofborg build openfortivpn-webview
